File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    stack = []
    for m in moves:
        for i in range(len(board)):
            if board[i][m-1] != 0:
                stack.append(board[i][m-1])
                board[i][m-1] = 0
                
                if len(stack) >= 2 and stack[-1]==stack[-2]:
                    stack.pop()
                    stack.pop()
                    answer += 2
                
                break
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    stack<int> stk;
    
    for (int idx : moves){
        for (int i=0; i<board.size(); i++){
            if (board[i][idx-1] != 0){
                if (!stk.empty() && board[i][idx-1] == stk.top()){
                    stk.pop();
                    answer += 2;
                }
                else
                    stk.push(board[i][idx-1]);
                board[i][idx-1] = 0;
                break;
            }
        }
    }
    return answer;
}